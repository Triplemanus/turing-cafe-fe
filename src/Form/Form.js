import React, { Component } from 'react';
import './Form.scss'

export default class Form extends Component {
  constructor() {
    super();
    this.state = {
      // id: '',
      name: '',
      date: '',
      time: '',
      number: ''
    }
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value})
  }

  submitNewRes = e => {
    e.preventDefault();
    const { addRes } = this.props;
    const newRes = { ...this.state, id: Date.now()};
    addRes(newRes);
    this.resetInputs();
  }

  resetInputs = () => {
    this.setState({ 
      name: '',
      date: '',
      time: '',
      number: ''
    })
  }

  render() {
    return (
      <form>
        <input
          type='text'
          placeholder='name'
          value={this.state.name}
          name='name'
          onChange={this.handleChange}
        />
        <input
          type='text'
          placeholder='date'
          value={this.state.date}
          name='date'
          onChange={this.handleChange}
        />
          <input
          type='text'
          placeholder='time'
          value={this.state.time}
          name='time'
          onChange={this.handleChange}
        />        <input
        type='text'
        placeholder='number'
        value={this.state.number}
        name='number'
        onChange={this.handleChange}
        />
        <button 
          onClick={this.submitNewRes}
        >
          Submit Reservation
        </button>
      </form>
    )
  }
}
