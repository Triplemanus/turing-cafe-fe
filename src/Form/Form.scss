input {
  border: 2px solid teal;
  height: 20px;
  width: 150px;
  padding: 4px;
  margin-right: 8px;
  border-radius: 5px;
  font-size: 16px;
  font-weight: 100;
  text-align: center;
}

.submit-button {
  border: 2px solid rgb(218, 219, 219);
  background-color: teal;
  color: rgb(241, 238, 238);
  height: 30px;
  width: 120px;
  border-radius: 5px;
  
}