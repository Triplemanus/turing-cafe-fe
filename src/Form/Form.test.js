import React from 'react';
import { shallow } from 'enzyme'
import Form from './Form';

describe('Form', () => {
  let wrapper;
  const mockAddReservation = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Form addRes={mockAddReservation} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state when handleChange is called', () => {
    const mockEvent = { target: {  name: 'name', value: "Frankenstein"} };
    const expected = "Frankenstein";

    wrapper.instance().handleChange(mockEvent);

    expect(wrapper.state('reservations')).toEqual(expected);
  });


})