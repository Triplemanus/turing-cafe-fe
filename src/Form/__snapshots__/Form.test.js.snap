// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<form>
  <input
    name="name"
    onChange={[Function]}
    placeholder="name"
    type="text"
    value=""
  />
  <input
    name="date"
    onChange={[Function]}
    placeholder="date"
    type="text"
    value=""
  />
  <input
    name="time"
    onChange={[Function]}
    placeholder="time"
    type="text"
    value=""
  />
          
  <input
    name="number"
    onChange={[Function]}
    placeholder="number"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Submit!
  </button>
</form>
`;
